server/controllers/api/user/password/index.js

```diff
@@ -0,0 +1,69 @@
+const { handleErrorResponse } = require('../../../utils/errorHandlers.js');
+const logger = require('winston');
+const db = require('../../../../models');
+const { auth: { masterPassword } } = require('../../../../../config/siteConfig.js');
+
+/*
+
+  route to update a password
+
+*/
+
+const updateUserPassword = ({ ip, originalUrl, body }, res) => {
+  let userRecord;
+  const { userName, oldPassword, newPassword } = body;
+
+  if (!masterPassword) {
+    return res.status(400).json({
+      success: false,
+      message: 'no master password set in site config',
+    });
+  }
+
+  if (!userName || !oldPassword || !newPassword) {
+    return res.status(400).json({
+      success: false,
+      message: 'body should include userName (channel name without the @), oldPassword, & newPassword',
+    });
+  }
+
+  db.User.findOne({
+    where: {
+      userName,
+    },
+  })
+  .then(user => {
+    userRecord = user;
```

Comment:
Why don't you just use `user` here?

Reply:
I need the `user` object to be available in the next `.then` in the chain, so I hoisted it up to the `updateUserPassword`'s scope by defining `userRecord` there.  Is there a better pattern?  I need the `user` object before I can check for a password match, and have to check for a password match before operating on that user object to change the password.

Reply:
Oh duh.